src/js/control-bar/skip-buttons/skip-forward.js

```diff
@@ -0,0 +1,67 @@
+import Button from '../../button';
+import Component from '../../component';
+
+/**
+ * Button to skip forward a configurable amount of time
+ * through a video. Renders in the control bar.
+ *
+ * e.g. options: {controlBar: {skipButtons: forward: 5}}
+ *
+ * @extends Button
+ */
+class SkipForward extends Button {
+  constructor(player, options) {
+    super(player, options);
+
+    this.validOptions = [5, 10, 30];
+    this.skipTime = this.getSkipForwardTime();
+
+    if (this.skipTime && this.validOptions.includes(this.skipTime)) {
+      this.show();
+    } else {
+      this.hide();
+    }
+  }
+
+  getSkipForwardTime() {
+    const playerOptions = this.options_.playerOptions;
+
+    return playerOptions.controlBar && playerOptions.controlBar.skipButtons && playerOptions.controlBar.skipButtons.forward;
+  }
+
+  buildCSSClass() {
+    return `vjs-skip-forward-${this.getSkipForwardTime()} ${super.buildCSSClass()}`;
+  }
+
+  /**
+   * On click, skips forward in the video by a configurable amount of seconds.
+   * If the time left in the video is less than the configured 'skip forward' time,
+   * skips to end of video.
+   *
+   * Handle a click on a `SkipForward` button
+   *
+   * @param {EventTarget~Event} event
+   *        The `click` event that caused this function
+   *        to be called
+   */
+  handleClick(event) {
+    const currentVideoTime = this.player_.currentTime();
+    const duration = this.player_.duration();
+
+    let newTime;
+
+    if (currentVideoTime + this.skipTime <= duration) {
```

Comment:
Live videos will have a duration of infinity. We shouldn't try to seek beyond the seekable range.

Reply:
Ahh I see, will make that change to use seekable range instead of duration and update the corresponding tests :)

Reply:
I've now added logic to check for seekable range if it is a live video (for both skip forward and skip backward buttons), and updated tests